doc/_extensions/zephyr/kconfig/__init__.py

```diff
@@ -429,8 +429,13 @@ def kconfig_build_resources(app: Sphinx) -> None:
 
         kconfig_db_file = outdir / "kconfig.json"
 
+        kconfig_db = {
+            "gh_base_url": app.config.gh_link_base_url,
```

Comment:
it should be its own config as otherwise it introduces an unwanted dependency between two otherwise independent Sphinx extensions
will require the config property to be declared in this extension's `setup()`, with `app.add_config_value(....)`
and yes, that means we have yet another extension declaring `something_something_baseurl = "https://github.com/zephyrproject-rtos/zephyr"` in conf.py :)